### PR TITLE
Use try-with-resource on FSMClient to avoid leak

### PR DESCRIPTION
**Pull request by StephenRi**
### What changes are proposed in this pull request?

fix usage for acquireMasterClientResource() in AlluxioFileInStream with try-with-resource

### Why are the changes needed?

resource leak

### Does this PR introduce any user facing changes?

none

### CODE REVIEW

**Comment by StephenRi** [comment]
@HelloHorizon I have send email to cla@alluxio.org, but no response. Is there anything wrong about my cla.

**Comment by HelloHorizon** [comment]
@StephenRi sorry about confusion, cla is well received

**Review by dbw9580 — APPROVED** [review]
LGTM, thanks!

File: core/client/fs/src/main/java/alluxio/client/file/AlluxioFileInStream.java [hunk 1]
```diff
@@ -227,7 +227,7 @@ public int read(ByteBuffer byteBuffer, int off, int len) throws IOException {
   // force a sync to update the latest metadata, then abort the current stream
   // The user should restart the stream and read the updated file
   private void refreshMetadataOnMismatchedLength(OutOfRangeException e) {
-    try {
+    try (CloseableResource<FileSystemMasterClient> client = mContext.acquireMasterClientResource()) {
```
Comment (dbw9580) [inline core/client/fs/src/main/java/alluxio/client/file/AlluxioFileInStream.java:230]:
```suggestion
    try (CloseableResource<FileSystemMasterClient> client =
        mContext.acquireMasterClientResource()) {
```

**Comment by StephenRi** [comment]
@dbw9580 Will the Integration Test of workflow rerun automatically if it failed?

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks!

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Comment by dbw9580** [comment]
cherry pick to 2.9? @jiacheliu3

**Comment by jiacheliu3** [comment]
This is a good-to-have for 2.9. The leak this fix catches only happens when all conditions match:
1. The underlying file in UFS is updated without Alluxio knowing
2. The update happens when the read stream is ongoing
3. Then Alluxio realizes the file inconsistency and triggers a forced metadata sync on this file
4. This single-file metadata sync throws an exception

That said, the leak is very unlikely to take place and even if it does, the consequence is loosing one client count (we have like 500 client counts available for each client process). So it's a mild leak to fix here.

I'll leave this up to your judgement @Xenorith whether this catches the 2.9 train. Thx!

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.9 please

**Pull request merged**